Source/WebCore/css/parser/CSSParserFastPaths.cpp

```diff
@@ -950,17 +933,37 @@ static RefPtr<CSSValue> parseColorWithAuto(StringView string, const CSSParserCon
 
 RefPtr<CSSValue> CSSParserFastPaths::maybeParseValue(CSSPropertyID propertyID, StringView string, const CSSParserContext& context)
 {
-    if (auto result = parseSimpleLengthValue(propertyID, string, context.mode))
-        return result;
-    if (propertyID == CSSPropertyDisplay)
+    switch (propertyID) {
```

Comment:
I would love to find a way to automate this, at least a bit, from CSSProperties.json. (it memory serves, I did a bit of that already for keywords, but this would be a bit more).

We would have to interrogate the grammar of a property to see if one of our fast path code paths works for it (at the very least, generating isSimpleLengthPropertyID() should be quite feasible).

Not for this change, but something I think is worth noodling over.